lucille-core/src/main/java/com/kmwllc/lucille/connector/storageclient/S3StorageClient.java

```diff
@@ -85,12 +85,17 @@ protected void traverseStorageClient(Publisher publisher, TraversalParams params
     response.stream()
         .forEachOrdered(resp -> {
           resp.contents().forEach(obj -> {
-            S3FileReference fileRef = new S3FileReference(obj);
-            processAndPublishFileIfValid(publisher, fileRef, params);
+            S3FileReference fileRef = new S3FileReference(obj, params);
+            processAndPublishFileIfValid(publisher, fileRef, params, state);
           });
         });
   }
 
+  @Override
```

Comment:
we should just reuse the same table name for all storage clients.  the file connector should specify where it stores its state so it knows where to find the state database.